packages-exp/auth-exp/src/platform_cordova/popup_redirect/popup_redirect.ts

```diff
@@ -64,6 +109,72 @@ class CordovaPopupRedirectResolver implements PopupRedirectResolver {
   ): void {
     throw new Error('Method not implemented.');
   }
+
+  private attachCallbackListeners(auth: Auth, manager: AuthEventManager): void {
+    const noEvent: AuthEvent = {
+      type: AuthEventType.UNKNOWN,
+      eventId: null,
+      sessionId: null,
+      urlResponse: null,
+      postBody: null,
+      tenantId: null,
+      error: _createError(AuthErrorCode.NO_AUTH_EVENT)
+    };
+
+    const noEventTimeout = setTimeout(async () => {
+      // We didn't see that initial event. Clear any pending object and
+      // dispatch no event
+      await _getAndRemoveEvent(auth);
+      manager.onEvent(noEvent);
+    }, INITIAL_EVENT_TIMEOUT_MS);
+
+    const universalLinksCb = async (
+      eventData: Record<string, string> | null
+    ): Promise<void> => {
+      // We have an event so we can clear the no event timeout
+      clearTimeout(noEventTimeout);
+
+      const partialEvent = await _getAndRemoveEvent(auth);
+      let finalEvent: AuthEvent | null = noEvent;
+      // Start with the noEvent
+      if (partialEvent && eventData?.['url']) {
+        finalEvent = _eventFromPartialAndUrl(partialEvent, eventData['url']);
+      }
+      manager.onEvent(finalEvent || noEvent);
+    };
+
+    // iOS 7 or 8 custom URL schemes.
+    // This is also the current default behavior for iOS 9+.
+    // For this to work, cordova-plugin-customurlscheme needs to be installed.
+    // https://github.com/EddyVerbruggen/Custom-URL-scheme
+    // Do not overwrite the existing developer's URL handler.
+    const existingHandleOpenUrl = window.handleOpenUrl;
+    window.handleOpenUrl = async url => {
```

Comment:
nit: what if developer try to set this field AFTER this statement, overwriting our handler?

Reply:
We can't control that. I'm not sure what the protocol is here but this is what the existing SDK does in any case: https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/cordovahandler.js#L802

Reply:
Fair enough.